python/test/unit/language/test_pipeliner.py

```diff
@@ -290,7 +290,7 @@ def test_pipeline_matmul(scale, device):
             # 2. check number of stages
             assert ttgir.count(f"num = {NUM_STAGES} : i32") != 0, "num_stages not match"
             # 3. check alloc
-            assert ttgir.count("ttg.local_alloc") == 2, "alloc number not match"
+            assert ttgir.count("ttg.local_alloc") != 0, "local_alloc not found"
```

Comment:
why is this changing?

Reply:
There was a third one getting added in TritonGPUReduceDataDuplication which I didn't think was a problem, but now I realize there was an opportunity that was getting missed when hoisting; I reverted this change and updated OptimizeDotOperands again with new tests as well to showcase the issue.